pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/design/SignatureDeclareThrowsExceptionRule.java

```diff
@@ -155,7 +155,9 @@ private boolean isAllowedMethod(ASTMethodDeclaration methodDeclaration) {
 
     @Override
     public Object visit(ASTConstructorDeclaration constructorDeclaration, Object o) {
-        checkExceptions(constructorDeclaration, o);
+        if (!junitImported || !getProperty(IGNORE_JUNIT_COMPLETELY_DESCRIPTOR)) {
```

Comment:
We used a slightly different pattern with multiple returns when checking method declaration. The equivalent for constructors would be:

```java
if (junitImported && getProperty(IGNORE_JUNIT_COMPLETELY_DESCRIPTOR)) {
    return super.visit(constructorDeclaration, o);
}

checkExceptions(constructorDeclaration, o);
return super.visit(constructorDeclaration, o);
```

I'm not sure which variant is better.... @jsotuyod ?

Reply:
I'm pro positive assertions rather than negatives...  but it's just a little of personal taste https://pmd.github.io/pmd/pmd_rules_java_codestyle.html#confusingternary